nvflare/private/fed/server/fed_server.py

```diff
@@ -403,14 +397,12 @@ def GetTask(self, request, context):
             self.logger.debug(f"Fetch task requested from client: {client.name} ({client.get_token()})")
             token = client.get_token()
 
-            # engine = fl_ctx.get_engine()
             shared_fl_ctx = fobs.loads(proto_to_bytes(request.context["fl_context"]))
             job_id = str(shared_fl_ctx.get_prop(FLContextKey.CURRENT_RUN))
-            # fl_ctx.set_peer_context(shared_fl_ctx)
+            run_processes = self.engine.get_run_processes()
```

Comment:
Move this run_processes = self.engine.get_run_processes() outside the "with self.lock:"  will loose the protection of the engine run_processes change. As the run_processes could be changed other job running thread.